### PR TITLE
Use query for removing packages

### PR DESCRIPTION
**Pull request by lasote**
Address #876

### CODE REVIEW

File: conans/client/remover.py [hunk 1]
```diff
@@ -75,65 +75,89 @@ def __init__(self, client_cache, search_manager, user_io, remote_proxy):
...
+    def _select_packages_to_remove(self, package_ids, has_remote, packages_query, pattern):
+        if has_remote:
+            if not packages_query:
+                conan_refs = self._remote_proxy.search(pattern)
+            else:
+                package_ids = self._remote_proxy.search_packages(pattern, packages_query).keys()
+                conan_refs = [pattern] if package_ids else None
+        else:
+            if not packages_query:
+                conan_refs = self._search_manager.search(str(pattern))
```
Comment (memsharded) [inline conans/client/remover.py]: Only one needing ``str()`` conversion?

Comment (memsharded) [inline conans/client/remover.py]: ``packages_ids`` input parameter, not used for logic, just returned? Sounds weird

File: conans/client/rest/rest_client.py [hunk 2]
```diff
@@ -309,7 +309,7 @@ def remove_conanfile(self, conan_reference):
     def remove_packages(self, conan_reference, package_ids=None):
         """ Remove any packages specified by package_ids"""
         self.check_credentials()
-        payload = {"package_ids": package_ids}
+        payload = {"package_ids": list(package_ids)}
```
Comment (memsharded) [inline conans/client/rest/rest_client.py]: if packages_ids is defaulted to None, weird, list(None) => raise Error

File: conans/client/remover.py (hunk 1 above)
Comment (lasote) [inline conans/client/remover.py]: About `packages_ids` what's your suggestion? ids are overwritten in case a packages_query exist or kept otherwise.

Comment (memsharded) [inline conans/client/remover.py]: I see... seemed a bit weird, but yes, other ways would be even worse. Can leave it as is.

**Pull request merged**